arch/posix/include/posix_soc_if.h

```diff
@@ -32,7 +32,20 @@ unsigned int posix_irq_lock(void);
 void posix_irq_unlock(unsigned int key);
 void posix_irq_full_unlock(void);
 int  posix_get_current_irq(void);
+#ifdef CONFIG_IRQ_OFFLOAD
 void posix_irq_offload(irq_offload_routine_t routine, void *parameter);
+#endif
+
+static ALWAYS_INLINE unsigned int z_arch_irq_lock(void)
+{
+	return posix_irq_lock();
+}
+
+
+static ALWAYS_INLINE void z_arch_irq_unlock(unsigned int key)
+{
+	posix_irq_unlock(key);
+}
```

Comment:
How come did you move these 2 back here? (it is not a deal breaker, but I would like to understand how come you decided to move those 2 into that file. And if need be place them in a more appropriate place)

Reply:
@aescolar I moved these here because the definitions of these functions need to be where posix_irq_lock() and posix_irq_unlock() are defined.

Otherwise I have a lot of problems because of mutual dependencies. include/arch/posix/arch.h implicitly pulls in this header, but this header also implicitly pulls in include/arch/posix/arch.h due to including irq_offload.h, and since the arch interface header has prototypes to inline functions they must always be defined.

Reply:
Uhm.. if you want to get this in as is, then just dismiss my review. But I would after, (myself if nobody else), untangle this.